paddle/fluid/framework/new_executor/interpretercore_util.cc

```diff
@@ -216,6 +216,171 @@ void apply_device_guard(const OperatorBase* op_base,
   }
 }
 
+// the return value is whether data transformer is needed for this var
+bool need_place_transform_for_var(const OpKernelType kernel_type_for_var,
+                                  const OpKernelType expected_kernel_key) {
+  if (platform::is_same_place(kernel_type_for_var.place_,
+                              expected_kernel_key.place_) ||
+      (is_cuda_pinned_place(kernel_type_for_var.place_) &&
+       is_cpu_place(expected_kernel_key.place_))) {
+    return false;
+  } else {
+    return true;
+  }
+}
+
+bool need_dtype_transform_for_var(const OpKernelType kernel_type_for_var,
+                                  const OpKernelType expected_kernel_key) {
+  return false;  // TODO(@xiongkun) add dtype judgement here)
```

Comment:
here) -> here, and I think throw error is better than `return false`, since it is not correct yet.